app.go

```diff
@@ -211,6 +211,15 @@ func New(opts ...Option) *App {
 	return app
 }
 
+// Error returns an error that may have been encountered during the
+// graph resolution.
+//
+// This includes things like incomplete graphs, circular dependencies,
+// missing dependencies, invalid constructors, etc.
+func (app *App) Error() error {
```

Comment:
I'd really prefer Err(), because Error() usually returns a string, implementing the error interface..